modules/@apostrophecms/rich-text-widget/ui/apos/components/AposRichTextWidgetEditor.vue

```diff
@@ -298,6 +291,54 @@ export default {
     stripPlaceholderBrs(html) {
       return html.replace(/<(p[^>]*)>\s*<br \/>\s*<\/p>/gi, '<$1></p>');
     },
+    // Legacy content may have `id` and `name` attributes on anchor tags
+    // but our tiptap anchor extension needs them on a separate `span`, so nest
+    // a span to migrate this content for each relevant anchor tag encountered
+    transformNamedAnchors(html) {
+      const el = document.createElement('div');
+      el.innerHTML = html;
+      const anchors = el.querySelectorAll('a[name]');
+      for (const anchor of anchors) {
+        const name = anchor.getAttribute('id') || anchor.getAttribute('name');
+        if (((typeof name) === 'string') && name.length) {
+          const span = document.createElement('span');
+          span.setAttribute('id', name);
+          anchor.removeAttribute('id');
+          anchor.removeAttribute('name');
+          if (anchor.children.length) {
+            // Migrate children of the anchor to the span
+            while (anchor.firstElementChild) {
+              span.append(anchor.firstElementChild);
+            }
+            anchor.prepend(span);
```

Comment:
could we [replace](https://developer.mozilla.org/en-US/docs/Web/API/Element/replaceWith) the old anchor instead of appending the new one in it?